serviceregistry-server/src/main/java/no/difi/meldingsutveksling/serviceregistry/servicerecord/ServiceRecordFactory.java

```diff
@@ -105,25 +111,35 @@ public ServiceRecord createEduServiceRecord(String orgnr) {
             serviceRecord.addDpeCapability(DPE_DATA.toString());
         }
 
-        return serviceRecord;
+        return Optional.of(serviceRecord);
     }
 
     @PreAuthorize("#oauth2.hasScope('move/dpe.read')")
-    private ServiceRecord createDpeServiceRecord(String orgnr) {
-        String pemCertificate = lookupPemCertificate(orgnr);
-        DpeServiceRecord sr = DpeServiceRecord.of(pemCertificate, orgnr);
-        if (elmaLookupService.identifierHasInnsynDataCapability(NORWAY_PREFIX + orgnr)) {
-            sr.addDpeCapability(DPE_DATA.toString());
+    public Optional<ServiceRecord> createDpeServiceRecord(String orgnr) {
+        if (elmaLookupService.identifierHasInnsynskravCapability(NORWAY_PREFIX + orgnr)) {
+            String pemCertificate;
+            try {
+                pemCertificate = lookupPemCertificate(orgnr);
+            } catch (CertificateNotFoundException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1166)